src/main/java/com/linkedin/kafka/clients/consumer/LiKafkaConsumerImpl.java

```diff
@@ -229,12 +229,12 @@ public void unsubscribe() {
       } catch (OffsetOutOfRangeException | NoOffsetForPartitionException oe) {
         switch (_offsetResetStrategy) {
           case EARLIEST:
-            _kafkaConsumer.seekToBeginning(oe.partitions());
             oe.partitions().forEach(_consumerRecordsProcessor::clear);
```

Comment:
Not sure I understand what you meant. This catch clause is to handle the case where broker returns an OffsetOutOfRangeException and the consumer has a offset reset strategy configured. In this case, reset the offset to either earliest and latest as if we haven't consumed any message.